src/python/datastore/data_types.py

```diff
@@ -942,8 +945,15 @@ def get_environment_string(self):
 
     return environment_string
 
+  def populate_indices(self):
+    """Populate keywords for fast test case list searching."""
+    self.keywords = list(
+        search_tokenizer.tokenize(self.name)
+        | search_tokenizer.tokenize(self.project))
+
   def _pre_put_hook(self):
     """Pre-put hook."""
+    self.populate_indices()
```

Comment:
One other thing: this needs to come after .project is set. Otherwise it could be empty.

Reply:
done